src/helpers/programHelper.js

```diff
@@ -800,6 +802,78 @@ class ProgramServiceHelper {
 
     return from(axios(req));
   }
+
+  /**
+   * Update the user profile with medium, subject and gradeLevel
+   *
+   * @param integer program_id  Program id
+   * @param integer user_id     User id
+   */
+  onAfterAddNomination(program_id, user_id) {
+    this.getProgramDetails(program_id).then(data => {
+      const value = {};
+      value['body'] = {
+        "id": "open-saber.registry.search",
+        "request": {
+            "entityType":["User"],
+            "filters": {
+              "userId": {
+                  "contains": user_id
+              }
+          }
+        }
+      };
+
+      const callback = (err, res) => {
+        if (!err && res) {
+          const user = _.first(res.data.result.User);
+          const updateRequestBody = {};
+          updateRequestBody['body'] = {
+            "id": "open-saber.registry.update",
+            "ver": "1.0",
+            "request": {
+              "User": {
+               "osid": user.osid,
+               "medium": _.union(user.medium, data.config.medium),
+               "gradeLevel": _.union(user.gradeLevel, data.config.gradeLevel),
+               "subject": _.union(user.subject, data.config.subject)
+               }
+            }
+          };
+          registryService.updateRecord(updateRequestBody, (error, response) => {
+            if (!error && response) {
+              return response;
+            } else {
+              return error;
+            }
+          });
+        }
+      };
+      registryService.searchRecord(value, callback);
+    });
+  }
+
+  /**
+   * Sort the program based on medium, subject, gradeLevel and program created date
+   *
+   * @param array  programs  List of programs
+   * @param object sort      Sort by options
+   */
+  sortPrograms(programs, sort) {
+    _.map(programs, program => {
+      program.matchCount =  _.intersection(JSON.parse(program.medium), sort.medium).length
+        + _.intersection(JSON.parse(program.gradeLevel), sort.gradeLevel).length
+        + _.intersection(JSON.parse(program.subject), sort.subject).length;
```

Comment:
In `_.intersection`, the order of arguments matter. Make sure you have checked that.